### PR TITLE
Fix for Husky Git Hooks

### LINKED ISSUE #17: Husky Git Hook Install
Turns out our seperate installs of [Husky](https://typicode.github.io/husky/#/?id=create-a-hook) for `frontend` and `backend` does not work. This is due to `husky install` registering with the root `.git` as a hook, resulting in only the most recently installed out of `fontend` or `backend` is eligible to receive commits.

iu

### PR DESCRIPTION
**Pull request by taitfuller**
As described in #17, our existing config for Husky was invalid as only one install of Husky is permitted per repo. As such, I have implemented the following changes:

- New `package.json` in the repository root with `devDependencies` `husky`, `lint-staged`, `eslint`, and `prettier`. This effectively moves and unifies the configuration from `frontend/` and `backend/` to the root.
- Removed dependencies of `husky` and `lint-staged` from `frontend/` and `backend/`
- New `.husky/` folder in the repository root that describes the `precommit` hook
- Removed `.husky/` folder from `frontend/` and `backend/`
- Setup a `prepare` script in `frontend/` and `/backend` that calls `cd .. && npm install` automatically whenever `npm install` is called within `frontend/` or `backend/`. This ensures that the `devDependencies` that implement the Git hook in the root are always installed alongside the `fontend/` and `backend/` without the user expressly having to call `npm install` from within the repository root.
- Added scripts `lint`, `lint:fix` and `format` to `frontend/`.
- Added scripts `test`, `lint`, `lint:fix` and `format` to the repository root.

Resolves #17.

### CODE REVIEW

**Review by Alxjlv — APPROVED** [review]
PR changes look solid to me

File: LICENSE.md [hunk 1]
```diff
@@ -6,115 +6,113 @@ Creative Commons Corporation (“Creative Commons”) is not a law firm and does
...
 Creative Commons public licenses provide a standard set of terms and conditions that creators and other rights holders may use to share original works of authorship and other material subject to copyright and certain other rights specified in the public license below. The following considerations are for informational purposes only, are not exhaustive, and do not form part of our licenses.
 
-* __Considerations for licensors:__ Our public licenses are intended for use by those authorized to give the public permission to use material in ways otherwise restricted by copyright and certain other rights. Our licenses are irrevocable. Licensors should read and understand the terms and conditions of the license they choose before applying it. Licensors should also secure all rights necessary before applying our licenses so that the public can reuse the material as expected. Licensors should clearly mark any material not subject to the license. This includes other CC-licensed material, or material used under an exception or limitation to copyright. [More considerations for licensors](http://wiki.creativecommons.org/Considerations_for_licensors_and_licensees#Considerations_for_licensors).
```
Comment (Alxjlv) [inline LICENSE.md:9]: Not sure how I feel about the license getting hit by prettier - this shouldn't be happening right?

File: README.md [hunk 2]
```diff
@@ -1 +1 @@
-# NerdsOpposeSociety
\ No newline at end of file
+# NerdsOpposeSociety
```
Comment (Alxjlv) [inline README.md:1]: Again, should .md files be getting hit by prettier?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "nerds-oppose-society",
+  "version": "0.1.0",
+  "scripts": {
+    "prepare": "husky install"
```
Comment (Alxjlv) [inline package.json:5]: We could also have lint and/or test script in here which runs the script for both (ie does `cd frontend && npm test && cd ../backend && npm test`)

**Review by Alxjlv — CHANGES_REQUESTED** [review]
Just some quick requests for comment

File: LICENSE.md (hunk 1 above)
Comment (taitfuller) [reply LICENSE.md:9]: I think it's fine - all it's actually changing is `__` to `**` and `___` to `**_` / `_**` ([equivalent styling from the Markdown docs](https://www.markdownguide.org/basic-syntax/#emphasis)) along with some whitespacing. Once again just keeping things consistent

File: README.md (hunk 2 above)
Comment (taitfuller) [reply README.md:1]: I think it's good to have - Prettier handles a bunch of different file types & it's nice to keep them consistent

File: package.json (hunk 3 above)
Comment (taitfuller) [reply package.json:5]: Yep sure can! Would you also like `start-backend` & `start-frontend` (or `start:backend` & `start:frontend`). Similar thing for `build`?

File: README.md (hunk 2 above)
Comment (Alxjlv) [reply README.md:1]: Okay I guess I'm all good with it!

File: package.json (hunk 3 above)
Comment (Alxjlv) [reply package.json:5]: Similar concepts - they're probably more of a nice to have though. Should be pretty easy to do at any rate

Comment (taitfuller) [reply package.json:9]: Cool I've implemented `test`, `lint`, `lint:fix` and `format` in c7db841

**Review by Alxjlv — APPROVED** [review]
Solid changes man, looks good

**Pull request merged**